src/Angor/Shared/ProtocolNew/Scripts/InvestmentScriptBuilder.cs

```diff
@@ -15,6 +15,13 @@ public InvestmentScriptBuilder(ISeederScriptTreeBuilder seederScriptTreeBuilder)
 
     public Script GetInvestorPenaltyTransactionScript(string investorKey, int punishmentLockDays)
     {
+        if (punishmentLockDays > 388)
+        {
+            // the actual number is 65535*512 seconds (388 days) 
+            // https://en.bitcoin.it/wiki/Timelock
+            throw new Exception($"Invalid CSV value {punishmentLockDays}");
```

Comment:
general question, wouldn't it better for a ArgumentOutOfRangeException ?

Reply:
yeah probably you are correct